lib/index.js

```diff
@@ -106,10 +107,19 @@ class GraphQLComponent {
     return check && check._types && check._resolvers && check._imports;
   }
 
-  execute(input, { root = undefined, context = {}, variables = {} } = {}) {
+  async execute(input, { root = undefined, context = {}, variables = {} } = {}) {
```

Comment:
This is kind of unrelated to the change being made here, but is sort of a, "while we are here" type of thing. I was working with Didier in his composition app and he pulled in a lower level component which uses a data source. The higher order component had a resolver that called execute() down to the lower order component and we were seeing undefined errors related to destructuring things off of context object in the resolver in the lower order component. Should we consider defaulting `context` to `this._context` instead of `{}` so that callers of `execute()` aren't required to pass the context object from the calling resolver?

Reply:
:+1: